django_graph_api/graphql/types.py

```diff
@@ -39,27 +39,45 @@ def __init__(self):
         Field.creation_counter += 1
 
     def get_value(self):
-        if hasattr(self.obj, 'get_{}'.format(self.name)):
-            return getattr(self.obj, 'get_{}'.format(self.name))()
+        raw_value = self.get_raw_value()
+        type = self.type_
+        if hasattr(type, 'coerce_result'):
+            try:
+                return type.coerce_result(raw_value)
+            except ValueError:
+                return None
+        return raw_value
+
+    def get_raw_value(self):
+        if hasattr(self.obj, 'get_{}'.format(self.selection_name)):
+            return getattr(self.obj, 'get_{}'.format(self.selection_name))()
 
         data = self.obj.data
         try:
-            return getattr(data, self.name)
+            return getattr(data, self.selection_name)
         except AttributeError:
             pass
 
         try:
-            return data.get(self.name)
+            return data.get(self.selection_name)
         except (AttributeError, KeyError):
             pass
 
         return None
 
     def bind(self, selection, obj):
         self.selection = selection
-        self.name = selection.name
+        self.selection_name = selection.name
```

Comment:
I had set it up as `self.name` to more closely mirror django model field APIs; I'm not necessarily against the change but I'm curious if there's a particular reason?

Reply:
Oh okay. I didn't look into what is getting passed in selection and I just know that when I was printing `self.name` on an IntegerField, it returned `number` which seemed odd to me. Also `Scalar` has a name property which returns its class name so I thought it would be more consistent that way. I'll change it back since it makes more sense for the Field API.